The new cast that redirects to `asType` is breaking a test that is casting a list to a Tuple.

```
Expected no exception to be thrown, but got 'nextflow.exception.ScriptCompilationException'
Expected no exception to be thrown, but got 'nextflow.exception.ScriptCompilationException'
	at spock.lang.Specification.noExceptionThrown(Specification.java:136)
	at nextflow.script.parser.v2.ScriptLoaderV2Test.should strip unsupported type annotations(ScriptLoaderV2Test.groovy:302)
Caused by: nextflow.exception.ScriptCompilationException: Script compilation failed
	at nextflow.script.parser.v2.ScriptLoaderV2.parse0(ScriptLoaderV2.groovy:129)
	at nextflow.script.parser.v2.ScriptLoaderV2.parse(ScriptLoaderV2.groovy:90)
	at nextflow.script.parser.v2.ScriptLoaderV2Test.should strip unsupported type annotations(ScriptLoaderV2Test.groovy:298)
Caused by: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
Main: 3: The class nextflow.script.types.Tuple<java.lang.String, java.lang.String, java.lang.String> (supplied with 3 type parameters) refers to the class nextflow.script.types.Tuple which takes no parameters
 @ line 3, column 44.
   '1', '1.fastq', '2.fastq'] as Tuple<Stri
```
                                 ^

--- a/modules/nextflow/src/test/groovy/nextflow/script/ScriptTypesTest.groovy
+++ b/modules/nextflow/src/test/groovy/nextflow/script/ScriptTypesTest.groovy
@@ -132,6 +132,23 @@ class ScriptTypesTest extends Dsl2Spec {
         samples[2].fastq == '3.fastq'
     }
 
+    def 'should strip unsupported type annotations' () {
+
+        when:
+        runScript(
+            '''\
+            // strip Tuple type from cast expression
+            ['1', '1.fastq', '2.fastq'] as Tuple<String,String,String>
+
+            // strip type annotation in variable declaration
+            def ch: Channel = channel.empty()
+            '''
+        )
+
+        then:
+        noExceptionThrown()
+    }
+
     def 'should replace instanceof on record type with runtime check' () {
 
         when:

--- a/modules/nextflow/src/test/groovy/nextflow/script/parser/v2/ScriptLoaderV2Test.groovy
+++ b/modules/nextflow/src/test/groovy/nextflow/script/parser/v2/ScriptLoaderV2Test.groovy
@@ -280,28 +280,6 @@ class ScriptLoaderV2Test extends Dsl2Spec {
         noExceptionThrown()
     }
 
-    def 'should strip unsupported type annotations' () {
-
-        given:
-        def session = new Session()
-        def parser = new ScriptLoaderV2(session)
-
-        def TEXT = '''
-            // strip cast type
-            ['1', '1.fastq', '2.fastq'] as Tuple<String,String,String>
-
-            // strip type annotation in variable declaration
-            def ch: Channel = channel.empty()
-            '''
-
-        when:
-        parser.parse(TEXT)
-        parser.runScript()
-
-        then:
-        noExceptionThrown()
-    }
-
     def 'should register outputs and topic emissions' () {
 
         given:

--- a/modules/nf-lang/src/main/java/nextflow/script/control/StripTypesVisitor.java
+++ b/modules/nf-lang/src/main/java/nextflow/script/control/StripTypesVisitor.java
@@ -113,13 +113,17 @@ public class StripTypesVisitor extends ClassCodeExpressionTransformer {
 
     private Expression transformCast(CastExpression node) {
         var type = node.getType();
+        if( STRIP_TYPES.contains(node.getType()) ) {
+            return node.getExpression();
+        }
         if( type.getGenericsTypes() != null ) {
             var fn = parameterizedType(type);
             return callThisX("_as_type", args(transform(node.getExpression()), classX(fn.getDeclaringClass()), constX(fn.getName())));
         }
-        else {
+        if( isNamedRecordType(type) ) {
             return callThisX("_as_type", args(transform(node.getExpression()), classX(type)));
         }
+        return node;
     }
 
     // Parameterized types in type casts are preserved from

